packages/fxa-settings/src/lib/storage-utils.ts

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Storage from './storage';
+
+export function isOriginalTab() {
+  const storage = Storage.factory('sessionStorage');
+  return storage.get('originalTab');
+}
+
+/* TODO. This runs in choose_what_to_sync, confirm_reset_password, confirm_signup_code,
+confirm, sign_in_token_code, and push/send_login if non-OAuth integration type. */
+export function persistVerificationData() {
```

Comment:
This + `unpersistVerificationData` will also be done in that refactor follow up. `persistOAuthVerificationData` can be done when we do OAuth stuff.

I thought I actually didn't need this for reset PW, but, then I realized we do indeed call it in `confirm_reset_password`.

The entire [afterVisible](https://github.com/mozilla/fxa/blob/a7708c4e8df572f0ec6e8e5c5247f47055d30e35/packages/fxa-content-server/app/scripts/views/confirm_reset_password.js#L65) portion of `confirm_reset_password` probably needs to be looked at and deals with same/different browser logic.